internal/datastore/postgres/postgres.go

```diff
@@ -127,8 +145,10 @@ func NewPostgresDatastore(
 func newPostgresDatastore(
 	ctx context.Context,
 	pgURL string,
+	replicaIndex int,
 	options ...Option,
 ) (datastore.Datastore, error) {
+	isPrimary := replicaIndex < 0
```

Comment:
I think this flag makes the rest of the function harder to follow. Maybe something like:

```go
newBasePostgresDatastore()

newPostgresDatastore(index) {
   ds := newBasePostgresDatastore()
   if index < 0 {
     return makeIntoRWDS(ds)
   }
   return makeIntoRODS(ds)
}
```

or
```go
newBasePostgresDatastore()

newReadWriteDatastore(base) {}

newReadReplicaDatastore(base) {}
```

Reply:
We still need the index for prom metrics and logs